transport/http.go

```diff
@@ -24,6 +24,18 @@ func (res httpRes) String() string {
 	return string(res.Body)
 }
 
+func HttpGet(client *http.Client, url string, headers map[string]string) (*httpRes, error) {
+	return httpDo(client, http.MethodGet, url, headers, nil)
+}
+
+func HttpPatch(client *http.Client, url string, data interface{}) (*httpRes, error) {
+	return httpDo(client, http.MethodPatch, url, nil, data)
+}
+
+func HttpPost(client *http.Client, url string, data interface{}) (*httpRes, error) {
+	return httpDo(client, http.MethodPost, url, nil, data)
+}
```

Comment:
I'm happy to include. I was worried these might be too opinionated (and also include fiocconfig logging style messages) that might not be good for other projects.

Wont httpRes wind up being an opaque value a client won't be able to use?

Reply:
> However, making httpRes public probably makes sense

I've added a new patch making the httpRes type public.